### PR TITLE
Fix `vertices(G)` to return all vertices, including isolated ones

### LINKED ISSUE #4426: Problems with `vertices(::Graph)` defined in AlgebraicStatistics
There is a problem with the `vertices(::Graph)` function in `experimental/AlgebraicStatistics/src/GraphicalModels.jl`:  
```
function vertices(G::Graph)
  E = [[src(e), dst(e)] for e in edges(G)]
  sort(unique(reduce(vcat, E)))
end
```

It implicitly assumes that every vertex is connected to an edge, which probably applies to all graphs in algebraic statistics, but need not hold in general.

@bkholler:  Would you mind renaming the function with a more appropriate name (e.g., `vertices_with_positive_degree`)?  

Of course, for the sake of consistency, it would be nice to have a general `vertices(::Graph)` function, but how that function should look like depends on whether vertex sets of polymake graphs need to be numbered `1` to `n` or whether there be gaps (like `{1,2,4}`).

### PR DESCRIPTION
**Pull request by benlorenz**
fixes #4426
cc: @YueRen, @bkholler 

The tests in algebraic statistics all seem to pass even with this change, so I don't think we need `vertices_with_positive_degree`.

### CODE REVIEW

File: test/Combinatorics/Graph.jl [hunk 1]
```diff
@@ -4,6 +4,7 @@
         g = Graph{Directed}(5)
         @test n_vertices(g) == 5
         @test n_edges(g) == 0
+        @test vertices(g) == 1:5
```
Comment (lgoettgens) [inline test/Combinatorics/Graph.jl:7]:
```suggestion
        @test issetequal(vertices(g), 1:5)
```
I think it would be better to compare these as sets, as the docstring does not guarantee the return type to be `UnitRange{Int}`.

Comment (benlorenz) [reply test/Combinatorics/Graph.jl:7]: This is not testing that the output is a `UnitRange`. Any `AbstractArray` would work, as long as the vertices are ordered correctly (and I think this is reasonable to require here even though it is not documented).

Edit: Testing with `issetequal` would allow vertices to return duplicates.

Comment (lgoettgens) [reply test/Combinatorics/Graph.jl:7]: I wasn't aware that there is a dispatch for that. Yes, this is fine then.

**Review by lgoettgens — APPROVED** [review]

**Comment by lgoettgens** [comment]
> The tests in algebraic statistics all seem to pass even with this change, so I don't think we need `vertices_with_positive_degree`.

IIUC, their graphs are all connected, and thus in particular have no isolated vertices.

**Comment by afkafkafk13** [comment]
> > The tests in algebraic statistics all seem to pass even with this change, so I don't think we need `vertices_with_positive_degree`.
> 
> IIUC, their graphs are all connected, and thus in particular have no isolated vertices.

Should there be one (small) testing case with an isolated vertex to prevent future regression? (Given that the authors did not come across the corner case by themselves, this might be a sensible idea.)

**Comment by benlorenz** [comment]
> > > The tests in algebraic statistics all seem to pass even with this change, so I don't think we need `vertices_with_positive_degree`.
> > 
> > 
> > IIUC, their graphs are all connected, and thus in particular have no isolated vertices.
> 
> Should there be one (small) testing case with an isolated vertex to prevent future regression? (Given that the authors did not come across the corner case by themselves, this might be a sensible idea.)

This new testcase for vertices has many isolated vertices:
https://github.com/oscar-system/Oscar.jl/blob/31c935c986b1582cc4ea3c70da56210f7d897d72/test/Combinatorics/Graph.jl#L4-L7

In Oscar 1.2.2:
```julia
julia> g = Graph{Directed}(5)
Directed graph with 5 nodes and no edges

julia> vertices(g)
ERROR: ArgumentError: reducing over an empty collection is not allowed; consider supplying `init` to the reducer
```

**Comment by afkafkafk13** [comment]
> This new testcase for vertices has many isolated vertices:
> 
> https://github.com/oscar-system/Oscar.jl/blob/31c935c986b1582cc4ea3c70da56210f7d897d72/test/Combinatorics/Graph.jl#L4-L7

Thank you! I had overlooked that one. No further comments from me.

**Review by afkafkafk13 — APPROVED** [review]
